packages/client/hmi-client/src/components/model/tera-model.vue

```diff
@@ -136,8 +141,11 @@ function onSaveAs() {
 }
 
 // Save modal
-function onModalSave() {
+function onModalSave(event: any) {
 	showSaveModal.value = false;
+	if (isWorkflow.value) {
+		emit('on-save', event);
+	}
```

Comment:
Why the test if we are in a workflow or not? What does this prevent to emit that event in the model asset page?